dl1_data_handler/image_mapper.py

```diff
@@ -1,744 +1,1112 @@
+            np.min(second_pos),
+            np.max(second_pos),
+            num=self.internal_shape * self.rebinning_mult_factor,
+            endpoint=True,
+        )
+        if len(self.x_ticks) < len(self.y_ticks):
+            x_grid, y_grid = np.meshgrid(grid_first, grid_second)
+        else:
+            x_grid, y_grid = np.meshgrid(grid_second, grid_first)
+        output_grid = np.column_stack([x_grid.ravel(), y_grid.ravel()])
+        return input_grid, output_grid
+
+    def _smooth_ticks(self, pix_pos, ticks):
+        """Smooth the ticks needed for the 'DigiCam' and 'CHEC' cameras."""
+        remove_val, change_val = [], []
+        for i in range(len(ticks) - 1):
+            if abs(ticks[i] - ticks[i + 1]) <= constants.tick_interval_limit:
+                remove_val.append(ticks[i])
+                change_val.append(ticks[i + 1])
+
+        ticks = [tick for tick in ticks if tick not in remove_val]
+        pix_pos = [
+            change_val[remove_val.index(pos)] if pos in remove_val else pos
+            for pos in pix_pos
+        ]
+        return pix_pos, ticks
+
+
+class SquareMapper(ImageMapper):
+    """
+    SquareMapper maps images to a square pixel grid without any modifications.
+
+    This class extends the functionality of ImageMapper by implementing
+    methods to generate a direct mapping table and perform the transformation.
+    It is particularly useful for applications where a direct one-to-one
+    mapping is sufficient for converting pixel data.for square pixel cameras
+    """
 
-        # We reshape each channel and then stack the result
-        result = []
-        for channel in range(n_channels):
-            vector = pixels[:, channel]
-            image_2d = (vector.T @ map_tab).reshape(
-                self.image_shapes[camera_type][0], self.image_shapes[camera_type][1], 1
+    def __init__(
+        self,
+        geometry,
+        config=None,
+        parent=None,
+        **kwargs,
+    ):
+        super().__init__(
+            geometry=geometry,
+            config=config,
+            parent=parent,
+            **kwargs,
+        )
+
+        if geometry.pix_type != PixelShape.SQUARE:
+            raise ValueError(
+                f"SquareMapper is only available for square pixel cameras. Pixel type of the selected camera is '{geometry.pix_type}'."
             )
-            result.append(image_2d)
-        telescope_image = np.concatenate(result, axis=-1)
-        return telescope_image
 
-    def get_indexmatrix(self, camera_type):
+        # Set shape of the image for the square camera
+        self.image_shape //= 2
+        self.internal_shape = self.image_shape
+
+        # Create square grid
+        input_grid, output_grid = self._get_square_grid()
+        # Calculate the mapping table
+        self.mapping_table = super()._generate_nearestneighbor_table(
+            input_grid, output_grid, pixel_weight=1.0
+        )
+
+    def _get_square_grid(
```

Comment:
Looks like this method has similar meaning to `get_grids()` of the other derived classes, or `get_grids_for_interpolation()` of the parent class, but has a different name. Consider a uniform name and an abstract method (or some sort of default implementation) in the base class.